optaplanner-core/src/main/java/org/optaplanner/core/api/domain/variable/VariableListener.java

```diff
@@ -79,4 +86,21 @@ default boolean requiresUniqueEntityEvents() {
      */
     void afterEntityRemoved(ScoreDirector<Solution_> scoreDirector, Entity_ entity);
 
+    /**
+     * Called when the entire working solution changes.
+     * At this point, implementations should clear state, if any.
+     * 
```

Comment:
This should specify if beforeEntityAdded(), afterEnityAdded() and/or beforeEntityRemoved()/afterEntityRemoved() are called too in such an event.
I don't recall if they actually are or aren't.